packages/govern-contract-utils/contracts/adaptative-erc165/AdaptativeERC165.sol

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity ^0.6.8;
+
+import "../erc165/ERC165.sol";
+
+contract AdaptativeERC165 is ERC165 {
+    // erc165 interface ID -> whether it is supported
+    mapping (bytes4 => bool) internal standardSupported;
+    // callback function signature -> magic number to return
+    mapping (bytes4 => bytes32) internal callbackMagicNumbers;
+
+    bytes32 internal constant UNREGISTERED_CALLBACK = bytes32(0);
+
+    event RegisteredStandard(bytes4 interfaceId);
+    event RegisteredCallback(bytes4 sig, bytes4 magicNumber);
+    event ReceivedCallback(bytes4 indexed sig, bytes data);
+
+    function supportsInterface(bytes4 _interfaceId) override virtual public view returns (bool) {
+        return standardSupported[_interfaceId] || super.supportsInterface(_interfaceId);
+    }
+
+    function _handleCallback(bytes4 _sig, bytes memory _data) internal {
+        bytes32 magicNumber = callbackMagicNumbers[_sig];
+        require(magicNumber != UNREGISTERED_CALLBACK, "adap-erc165: unknown callback");
+
+        emit ReceivedCallback(_sig, _data);
+
+        // low-level return magic number
+        assembly {
+            mstore(0x00, magicNumber)
+            return(0x00, 0x20)
+        }
```

Comment:
Why we need to use assembly here?

Reply:
Because we need to do a low-level return (have a fallback return stuff)